tests/test_tempfile.py

```diff
@@ -0,0 +1,185 @@
+from __future__ import annotations
+
+import os
+import pathlib
+import shutil
+import tempfile
+
+import pytest
+
+from anyio import (
+    NamedTemporaryFile,
+    SpooledTemporaryFile,
+    TemporaryDirectory,
+    TemporaryFile,
+    gettempdir,
+    gettempdirb,
+    gettempprefix,
+    gettempprefixb,
+    mkdtemp,
+    mkstemp,
+    to_thread,
+)
+
+pytestmark = pytest.mark.anyio
+
+
+@pytest.mark.anyio
+async def test_temporary_file() -> None:
+    data = b"temporary file data"
+    async with TemporaryFile[bytes]() as af:
+        await af.write(data)
+        await af.seek(0)
+        result = await af.read()
+    assert result == data
```

Comment:
Always add a blank line between the end of a control block and the code that follows.
```suggestion

    assert result == data
```